qiskit/providers/ibmq/ibmqjob.py

```diff
@@ -281,7 +303,22 @@ def status(self):
         return self._status
 
     def error_message(self):
-        """Return the error message returned from the API server response."""
+        """Provide details about the reason of failure.
+
+        Calling this method if the job status is ERROR, consumes the job.
+
+        Returns:
+            str: An error report if the job errored or None elsewhere.
+        """
+        self._wait_for_completion()
+        if self.status() is not JobStatus.ERROR:
+            return None
+
+        job_response = self._get_job()
+        results = job_response['qObjectResult']['results']
+        if not self._api_error_msg:
```

Comment:
Can you move this `if` block higher (before `_get_job()`) and adjust accordingly (as in, probably including the previous two lines in an `else` or similar)? Otherwise it seems it will attempt to consume the job twice, and the function might error out at that point ungracefully.